src/service/history-impl.js

```diff
@@ -1033,7 +1041,11 @@ export class HistoryBindingVirtual_ {
     if (data['newStackIndex'] !== undefined) {
       data['stackIndex'] = data['newStackIndex'];
     }
-    this.updateHistoryState_(/** @type {!HistoryStateDef} */ (data));
+    if (this.isHistoryState_(data)) {
```

Comment:
Is this the only real change in the file? Everything else is for debugging?

Reply:
Also minor semantic change in `isHistoryState_` to avoid `state.stackIndex: 0` being labeled as invalid.

Before:
```js
maybeHistoryState && maybeHistoryState['stackIndex']
```

After:
```js
maybeHistoryState && maybeHistoryState['stackIndex'] !== undefined
```